File: back/onboarding/src/main/java/com/dev/op/core/service/vipchannel/StorageService.java
package com.dev.op.core.service.vipchannel;

import java.util.List;

import com.dev.op.core.dto.ResponseModel;
import com.dev.op.core.dto.vipchannel.getListRemisionByIdModel;
import com.dev.op.core.dto.vipchannel.getListRemisionModel;
import com.dev.op.core.dto.vipchannel.getProdctModel;
import com.dev.op.core.dto.vipchannel.getProductProviderModel;
import com.dev.op.core.dto.vipchannel.getProviderModel;

public interface StorageService {
	List<getProdctModel> getProdct();
	List<getProviderModel> getProvider();
	List<getListRemisionModel> getListRemision();
	List<getProductProviderModel> getProductProvider();
	List<getListRemisionByIdModel> getListRemisionById(Integer remisId);
	List<ResponseModel> postProductP(Integer productId,Integer provideId,String price);
	List<ResponseModel> postSaveProvide(String name);
	List<ResponseModel> postSaveProduc(String name,String code,String description,String codeP,String medi);
	List<ResponseModel> postRemision(Integer options,Integer typeRemi,Integer producId,Integer providerId,Integer counts,String conditions,String autorize,Integer codeuser);
}

File: front/onboarding/src/app/services/storage.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { AlmacenModel, ProductoModel, ProveedorModel, ProductoProveedorModel } from '../models/storage.model';
import { ResponseModel } from '../models/personpay.model';
import { environment } from '../../environments/environment';

@Injectable({
    providedIn: "root"
})

export class StorageService {
    results: AlmacenModel[];

    constructor(private httpClient: HttpClient) { }

    httpOptions = {
        headers: new HttpHeaders({
            
        })
      }

    private BASE_URL: string = environment.serverLocal;

    listarproducto(): Observable<ProductoModel[]> {
        return this.httpClient.get<ProductoModel[]>(`${this.BASE_URL}/almacen/getProdct`);
    }

    guardarProducto(name: string,code:string,description:string,codeP:number,medi:string): Observable<ResponseModel[]> {
        return this.httpClient.get<ResponseModel[]>(`${this.BASE_URL}/almacen/postSaveProduc/`+
        name + "/" + code + "/" + description + "/" + codeP + "/" + medi);
    }

    editarProducto(producId:number,name: string,code:string,description:string,codeP:number,medi:string): Observable<ResponseModel[]> {
        return this.httpClient.get<ResponseModel[]>(`${this.BASE_URL}/almacen/putSaveProduct/`+
        producId + "/" +name + "/" + code + "/" + description + "/" + codeP + "/" + medi);
    }

    listarproveedor(): Observable<ProveedorModel[]> {
        return this.httpClient.get<ProveedorModel[]>(`${this.BASE_URL}/almacen/getProvider`);
    }

    guardarProveedor(name: string): Observable<ResponseModel[]> {
        return this.httpClient.get<ResponseModel[]>(`${this.BASE_URL}/almacen/postSaveProvide/`+ name);
    }

    editarProoveedor(provideId:number,name: string): Observable<ResponseModel[]> {
        return this.httpClient.get<ResponseModel[]>(`${this.BASE_URL}/almacen/putSaveProvide/`+name + "/" +provideId);
    }

    listarproveedorproducto(optio:number,productoE:number): Observable<ProductoProveedorModel[]> {
        return this.httpClient.get<ProductoProveedorModel[]>(`${this.BASE_URL}/almacen/getProductProvider`
        + "/" + optio + "/" + productoE);
    }

    edita(productId:number,provideId: number,price: string): Observable<ResponseModel[]> {
        return this.httpClient.get<ResponseModel[]>(`${this.BASE_URL}/almacen/postProductP/`
        +productId + "/" +provideId+ "/" +price);
    }

    guardar(){

    }

    listarremisiones(): Observable<AlmacenModel[]> {
        return this.httpClient.get<AlmacenModel[]>(`${this.BASE_URL}/almacen/getListRemision`);
    }

    listarremisionesById(id:number): Observable<AlmacenModel[]> {
        return this.httpClient.get<AlmacenModel[]>(`${this.BASE_URL}/almacen/getListRemisionById/` + id);
    }

}